File: bug02_missing_quotation_mark/Main.java
import java.nio.ByteBuffer;
import java.util.Base64;

public class Main {
    public static long stringToLong(String str) {
        byte[] bytes = str.getBytes();
        long val = 0;
        for (byte b : bytes) {
            val = (val << 8) | (b & 0xFF);
        }
        return val;
    }

    public static long build_hash(String[] parts) {
        long acc = 0;
        for (int idx = 0; idx < parts.length; idx++) {
            long val = stringToLong(parts[idx]);
            acc = (acc ^ val) * 0x9E3779B97F4A7C15L;
            int shift = (idx % 7) + 1;
            acc = (acc << shift) | (acc >>> (64 - shift));
        }
        return acc;
    }

    public static void main(String[] args) {
        String[] tokens = {"alpha", "bravo, "charlie", "delta", "echo",
                           "foxtrot", "golf", "hotel", "india", "juliet"};
        long result = build_hash(tokens);
        ByteBuffer buffer = ByteBuffer.allocate(Long.BYTES);
        buffer.putLong(result);
        String encoded = Base64.getEncoder().encodeToString(buffer.array());
        System.out.println(encoded);
    }
}

File: bug02_missing_quotation_mark/main.js
function stringToBigInt(str) {
    let buf = Buffer.from(str, 'utf8');
    let val = 0n;
    for (let i = 0; i < buf.length; i++) {
        val = (val << 8n) | BigInt(buf[i]);
    }
    return val;
}

function build_hash(parts) {
    let acc = 0n;
    for (let idx = 0n; idx < parts.length; idx++) {
        let val = stringToBigInt(parts[idx]) & 0xFFFFFFFFFFFFFFFFn;
        acc = (acc ^ val) * 0x9E3779B97F4A7C15n;
        acc &= 0xFFFFFFFFFFFFFFFFn;
        
        let shift = (idx % 7n) + 1n;
        acc = ((acc << shift) | (acc >> (64n - shift))) & 0xFFFFFFFFFFFFFFFFn;
    }
    return acc;
}

let tokens = ["alpha", "bravo, "charlie", "delta", "echo",
              "foxtrot", "golf", "hotel", "india", "juliet"];
let result = build_hash(tokens);

let buffer = Buffer.alloc(8);
buffer.writeBigUInt64BE(result);
console.log(buffer.toString('base64'));
